homeassistant/components/sensor/__init__.py

```diff
@@ -246,6 +252,23 @@ def device_class(self) -> SensorDeviceClass | None:
             return self.entity_description.device_class
         return None
 
+    @final
+    @property
+    def is_numeric(self) -> bool:
+        """Return true if the sensor is numeric."""
+        if (
+            self.state_class is not None
+            or self.native_unit_of_measurement is not None
+            or self.native_precision is not None
+        ):
+            return True
+        with suppress(ValueError):
```

Comment:
It looks like device_class can be unbound now... sorry about that.
```suggestion
        device_class: SensorDeviceClass | None = None
        with suppress(ValueError):
```